doc/nvim-treesitter.txt

```diff
@@ -222,7 +222,112 @@ Supported options:
       enable = true
     },
   }
+
+`@indent`				    *nvim-treesitter-indentation-queries*
+Queries can use the following captures: `@indent` and `@dedent`,
+`@branch`, `@indent_end` or `@aligned_indent`. An `@ignore` capture tells
+treesitter to ignore indentation and a `@zero_indent` capture sets
+the indentation to 0.
+
+`@indent`				    *nvim-treesitter-indentation-indent*
+The `@indent` specifies that the next line should be indented.  Multiple
+indents on the same line get collapsed. Eg.
+
+>
+  (
+   (if_statement)
+   (ERROR "else") @indent
+  )
 <
+Indent can also have `immediate_indent` set using a `#set!` directive, which
+permits the next line to indent even when the block intended to be indented
+has no content yet, improving interactive typing.
+
+eg for python:
+>
+ ((if_statement) @indent
+   (#set! "immediate_indent" 1))
+<
+
+Will allow:
+>
+  if True:<CR>
+      # Auto indent to here
+
+`@indent_end`				*nvim-treesitter-indentation-indent_end*
+An `@indent_end` capture is used to specify that the indented region ends and
+any text subsequent to the capture should be dedented.  
+
+`@branch`				*nvim-treesitter-indentation-branch*
+An `@branch` capture is used to specify that a dedented region starts
+at the line including the captured nodes.
+
+`@dedent`				*nvim-treesitter-indentation-dedent*
+A `@dedent` capture specifies dedenting starting on the next line.
+>
+`@aligned_indent`			*nvim-treesitter-indentation-aligned_indent*
+Aligned indent blocks may be specified with the `@aligned_indent` capture.
+This permits
+
+>
+  foo(a,
+      b,
+      c)
+<
+As well as
+>
+  foo(
+    a,
+    b,
+    c)
+<
+and finally
+>
+  foo(
+    a,
+    b,
+    c
+  )
+<
+To specify the delimiters to use `open_delimiter` and `close_delimiter`
+should be used.  Eg.
+>
+ ((argument_list) @aligned_indent
+  (#set! "open_delimiter" "(")
```

Comment:
When I said to separate these I had this in mind:

```
(#set! indent.delimeters "(" ")")
```

Reply:
oh, ok.. I will rework then

Reply:
Can I understand a bit more. 

Does 

```(#set! "Param" "(" ")")```

Work right now? A cursory reading of the query.lua implementation indicates it does not and that there can be only one value to set. 

Secondly, the location to set is unquoted and has a period in the path. Does that work right now? 

Is the ask that it would be 

```(#set! aligned_indent.delimiters "(" ")")```

Or

```(#set! indent.delimiters "(" ")")```

If the latter I don't quite understand as this is relevant to aligned_indent only.

Reply:
Yeah, ```runtime/lua/vim/treesitter/query.lua``` (see https://github.com/neovim/neovim/blob/master/runtime/lua/vim/treesitter/query.lua#L422)  has the set directive parser either accept ```(#set! @capture "key" "value")``` or ```(#set! "key" "value")``` as documented in the help.  Options are to revert back to having a single ```delimiters``` with a string assumed to be two chars long, or keep as-is.  I'm not sure how to do anything more than that without additional upstream changes.

Reply:
I'd keep both set opening and closing delimiter (for now), as it works and is more easily extendable for the near future.
Besides, once `#set!` supports setting multiple values, it would be easy to change the code to accomodate for the changes.

Reply:
I just reverted that chunk of changes in how delimiters are specifed - so keeps the diff small.  It isn't the point of this PR, and I would if needed be happy to apply that as a follow on.  I did this before I saw your comment annoyingly.  But maybe it's still the right thing to do and apply separating them out as a separate PR.

Reply:
If the sentiment is strong to have them split, it's easy to do so. But I can do so as a clear separate commit to view just those diffs, with the others squashed.

Reply:
Cause we changing the `#set!` to different delimiters is easier than asking upstream to support `#set!` changing to list of values, it's a really big breaking change (I believe).

Reply:
ok, I brought that back and the docs still reflect it.  So it is 

```
(#set! "open_delimiter" "{")
(#set! "close_delimiter" "}")
```

Just that diff is a separate commit so it's clear which is doing what.

Reply:
Ok to keep them as separate directives for now, but please try and scope all captures and set keys into `indent.*`. And no, I don't believe quotes should not be necessary.

```
(#set! indent.delimiter.open "{")
(#set! indent.delimiter.close "}")
```

The `set` directives are completely implemented by us and `nvim-treesitter` can override the implementation if it wishes (until upstream can update).

Our queries aren't being shared right now, so what's important at this stage is to design a good system that's coherent with other query schemes.